src/LinkedListTest.java

```diff
@@ -26,21 +54,38 @@ void testAddNotFirstElement() throws Exception {
         assertEquals(node1.getNextElement(), node2);
     }
 
+    @Test
+    void testReverse() throws Exception{
+        LinkedList list = new LinkedList();
+        list.addElement(new Node(0));
+        list.addElement(new Node(1));
+        list.addElement(new Node(2));
+        list.addElement(new Node(3));
+        LinkedList reverse = new LinkedList();
+        reverse=list.reverse();
+        if (list.getLengthOfList()==0){
+            assertEquals(reverse.getHead(),null);
+        }else if(list.getLengthOfList()==1){
+            assertEquals(reverse.getHead(),list.getHead());
+        }else if((list.getLengthOfList() % 2)==0 && list.getLengthOfList()!=1){
+            for(int i=list.getLengthOfList()-1;i>=0;i--){
+                assertEquals(reverse.getElementById(reverse.getLengthOfList()-i-1).getData(),list.getElementById(i).getData());
+            }
+        }else {
```

Comment:
what the hell is that?? Empty cases? 
if you have 'spaghetti' in the test, you are, probably, doing wring things...